main.cpp

```diff
@@ -42,7 +42,112 @@ send me a DM to check your pull request
  */
 
 #include <iostream>
+
+struct FloatType
+{
+    float add(float lhs, float rhs)
+    {
+        return lhs + rhs;
+    }
+
+    float subtract(float lhs, float rhs)
+    {
+        return lhs - rhs;
+    }
+
+    float multiply(float lhs, float rhs)
+    {
+        return lhs * rhs;
+    }
+
+    float divide(float lhs, float rhs)
+    {
+        if(rhs == 0.f) std::cout << "Attention: Dividing floats by 0!" << "\n";
+        return lhs / rhs;
+    }
+};
+
+struct DoubleType
+{
+    double add(double lhs, double rhs)
+    {
+        return lhs + rhs;
+    }
+
+    double subtract(double lhs, double rhs)
+    {
+        return lhs - rhs;
+    }
+
+    double multiply(double lhs, double rhs)
+    {
+        return lhs * rhs;
+    }
+
+    double divide(double lhs, double rhs)
+    {
+        if(rhs == 0.0) std::cout << "Attention: Dividing doubles by 0!" << "\n";
+        return lhs / rhs;
+    }
+};
+
+struct IntType
+{
+    int add(int lhs, int rhs)
+    {
+        return lhs + rhs;
+    }
+
+    int subtract(int lhs, int rhs)
+    {
+        return lhs - rhs;
+    }
+
+    int multiply(int lhs, int rhs)
+    {
+        return lhs * rhs;
+    }
+
+    int divide(int lhs, int rhs)
+    {
+        if(rhs == 0) 
+        {
+            std::cout << "ERROR: Dividing ints by 0!" << "\n";
+            return 0;
+        }
+        else return lhs / rhs;
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
if the `if()` block is entered, the `return aValue;` is the last line of code to be executed.
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
`return` statements are the last thing that ever happens in a function
That means if your `if()` block is NOT entered, it will be skipped.  What is the next thing that happens?  `return anotherValue;`
that means you do NOT need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    return anotherValue;
}
```